README.md

```diff
@@ -1 +1,569 @@
+
+```Go
+import "xyz"
+
+const route = "/some-route"
+
+var NotFoundErr = errors.New("not found")
+
+func someFunc() {
+	//...
+}
+
+func someOtherFunc() {
+	// usage of route
+}
+
+func yetAnotherFunc() {
+	// usage of NotFoundErr
+}
+```
+
+## Avoid side-effects
+
+Bad:
+
+```Go
+func init() {
+	someStruct.Load()
+}
+```
+
+Side effects are only okay in special cases (e.g. parsing flags in a cmd). If you find no other way, rethink and refactor.
+
+## Don't over-interface
+
+Bad:
+
+```Go
+type Server interface {
+	Serve() error
+	Some() int
+	Fields() float64
+	That() string
+	Are([]byte) error
+	Not() []string
+	Necessary() error
+}
+```
+
+Good:
+
+```Go
+type Server interface {
+	Serve() error
+}
+
+func run(srv Server) {
+	srv.Serve()
+}
+```
+
+Favour small interfaces and only expect the interfaces you need in your funcs.
+
+## Goimports
+
+https://godoc.org/golang.org/x/tools/cmd/goimports
+
+Command goimports updates your Go import lines, adding missing ones and removing unreferenced ones. In addition to fixing imports, goimports also formats your code in the same style as gofmt so it can be used as a replacement for your editor's gofmt-on-save hook.
+
+Run gofmt on your code to automatically fix the majority of mechanical style issues.
+
+## Comment Sentences
+Comments documenting declarations should be full sentences, even if that seems a little redundant. Comments should begin with the name of the thing being described and end in a period:
+
+```Go
+// Request represents a request to run a command.
+type Request struct { ...
+
+// Encode writes the JSON encoding of req to w.
+func Encode(w io.Writer, req *Request) { ...
+```
+
+## Declaring Empty Slices
+
+When declaring an empty slice, prefer
+
+```Go
+var t []string
+```
+
+over
+
+```Go
+t := []string{}
+```
+
+The former declares a nil slice value, while the latter is non-nil but zero-length. They are functionally equivalent—their len and cap are both zero—but the nil slice is the preferred style.
+
+Note that there are limited circumstances where a non-nil but zero-length slice is preferred, such as when encoding JSON objects (a nil slice encodes to null, while []string{} encodes to the JSON array []).
+
+## Doc Comments
+
+All top-level, exported names should have doc comments, as should non-trivial unexported type or function declarations.
+See https://golang.org/doc/effective_go.html#commentary for more information about commentary conventions.
+
+## Don't Panic
+
+Don't use panic for normal error handling. Use error and multiple return values.
+See https://golang.org/doc/effective_go.html#errors for more information.
+
+## Exports
+
+Do not export functions and variables unless necessary.
+
+Keep the exported variables and error in the same package that is using/returning them.
+For example i/o error like `io.EOF` from the go std lib is in the io package, where the function returning this error are also defined.
+
+## Errors
```

Comment:
add section about logging and logging errors. 
Add info about structured logging (logrus) and correlation IDs (?)

Reply:
added: https://github.com/apaliavy/golangcodestyle/blob/codestyle-proposal/README.md#logging
@szank Please improve this section with additional details if you can :)

Reply:
So, I would add something like this: 
``` 
Logging: 
We should be logging the correlation ID in our logs wherever possible  (that is when the flow is initiated by an user action (http/grpc call). 
To do this we need to use function: 
LoggerFromContext from github.com/karhoo/lib-common/log package. 
The logger object should be passed to the child functions as a function argument. If the function needs to add more tags to the log, it can use WithFields function. The function will return a copy of the logger, so the parent logger will not be modified when the function returns. 
```
Please feel free to redact it. Cannot see the context here. The main point is using `LoggerFromContext` and passing the logger down the call tree as an argument.

Reply:
updated!